modules/phase_field/src/materials/SphericalGaussianMaterial.C

```diff
@@ -0,0 +1,632 @@
+{
+  std::ifstream File1(_quaternion_file_name.c_str());
+  // Account for opening error
+  if (!File1.is_open())
+    mooseError("Unable to open quaternion file");
+
+  Real quat0, quat1, quat2, quat3;
+  while (File1 >> quat0 >> quat1 >> quat2 >> quat3)
+  {
+    // Calculate the quaternion norm
+    ADReal qcomp0 = quat0;
+    ADReal qcomp1 = quat1;
+    ADReal qcomp2 = quat2;
+    ADReal qcomp3 = quat3;
+    ADReal qcomp_norm =
+        std::sqrt(qcomp0 * qcomp0 + qcomp1 * qcomp1 + qcomp2 * qcomp2 + qcomp3 * qcomp3);
+    // Normalize the quaternion components
+    ADReal qcomp0_normalized = qcomp0 / qcomp_norm;
+    ADReal qcomp1_normalized = qcomp1 / qcomp_norm;
+    ADReal qcomp2_normalized = qcomp2 / qcomp_norm;
+    ADReal qcomp3_normalized = qcomp3 / qcomp_norm;
+    // Store the normalized quaternion components
+    _q0.push_back(qcomp0_normalized);
+    _q1.push_back(qcomp1_normalized);
+    _q2.push_back(qcomp2_normalized);
+    _q3.push_back(qcomp3_normalized);
+  }
+
+  File1.close();
+}
+
+// Read and store minima misorientations (minima library) data from text file
+void
+SphericalGaussianMaterial::readLibraryFile()
+{
+  std::ifstream File2(_library_file_name.c_str());
+  // Account for opening error
+  if (!File2.is_open())
+    mooseError("Unable to open library file");
+
+  Real lib1, lib2, lib3, lib4, lib5, lib6, lib7, lib8, lib9;
+  while (File2 >> lib1 >> lib2 >> lib3 >> lib4 >> lib5 >> lib6 >> lib7 >> lib8 >> lib9)
+  {
+    // Create vectors and normalize: Axis of rotation and Gaussian direction for library
+    // misorientation
+    ADRealVectorValue library_rotation_axis_value(lib1, lib2, lib3);
+    ADRealVectorValue library_gaussian_direction_value(lib5, lib6, lib7);
+    library_rotation_axis_value = library_rotation_axis_value.unit();
+    library_gaussian_direction_value = library_gaussian_direction_value.unit();
+    // Verify that grain boundary energy and pre-determined value of L for library misorientation
+    // are positive
+    if (lib8 <= 0)
+      mooseError("library_minima_energy must be positive. Found value: ", lib8);
+    if (lib9 <= 0)
+      mooseError("library_minima_L must be positive. Found value: ", lib9);
+    // Store the values in sequence
+    _library_rotation_axis.push_back(library_rotation_axis_value);
+    _library_rotation_angle.push_back(lib4);
+    _library_gaussian_direction.push_back(library_gaussian_direction_value);
+    _library_minima_energy.push_back(lib8);
+    _library_minima_L.push_back(lib9);
+  }
+
+  File2.close();
+}
+
+// Compute material properties and derivatives
+void
+SphericalGaussianMaterial::computeQpProperties()
+{
+  // Determine number of grains
+  _grain_num = _grain_tracker.getTotalFeatureCount();
+  // Determine order parameters to grain index
+  const auto & op_to_grains = _grain_tracker.getVarToFeatureVector(_current_elem->id());
+
+  // Initialize values
+  ADReal aniso_epsilon = 0.0;
+  ADRealVectorValue anisoplus_epsilon(0.0, 0.0, 0.0);
```

Comment:
`ADRealVectorValue` is automatically initialized to zero. So all those could be 
```suggestion
  ADRealVectorValue anisoplus_epsilon;
```